internal/operands/common-instancetypes/reconcile.go

```diff
@@ -270,33 +274,58 @@ func (c *CommonInstancetypes) reconcileFromURL(request *common.Request) ([]commo
 }
 
 func (c *CommonInstancetypes) reconcileFromBundle(request *common.Request) ([]common.ReconcileResult, error) {
+	// Handle the featureGate being disabled by ensuring any resources previously reconciled are cleaned up
+	if !isFeatureGateEnabled(request) {
+		return c.cleanupReconcile(request)
+	}
+
 	request.Logger.Info("Reconciling common-instancetypes from internal bundle")
-	clusterInstancetypesFromBundle, clusterPreferencesFromBundle, err := c.fetchResourcesFromBundle()
+	var err error
+	c.virtualMachineClusterInstancetypes, c.virtualMachineClusterPreferences, err = c.fetchResourcesFromBundle()
 	if err != nil {
 		return nil, err
 	}
 
 	// Remove any resources no longer provided by the bundle
-	if err = c.reconcileRemovedResources(request, clusterInstancetypesFromBundle, clusterPreferencesFromBundle); err != nil {
+	if err = c.reconcileRemovedResources(request); err != nil {
 		return nil, err
 	}
 
-	c.virtualMachineClusterInstancetypes = clusterInstancetypesFromBundle
-	c.virtualMachineClusterPreferences = clusterPreferencesFromBundle
 	reconcileFuncs, err := c.reconcileFuncs(request)
 	if err != nil {
 		return nil, err
 	}
 	return common.CollectResourceStatus(request, reconcileFuncs...)
 }
 
+func isFeatureGateEnabled(request *common.Request) bool {
+	if request.Instance.Spec.FeatureGates == nil || request.Instance.Spec.FeatureGates.DeployCommonInstancetypes == nil {
```

Comment:
If those fields are `nil` and we consider that InstanceTypes should be deployed, it is because we want to deploy them by default. Am I right? Is this intentional?

Reply:
Yes, by default they should still be deployed.

Reply:
ACK, thanks for the clarification.